### PR TITLE
jackal_base package

### PR DESCRIPTION
**Pull request by mikepurvis**
This has sat on a branch long enough, causing confusion. The jackal_base package is primarily a place for `jackal_node` to live, which does the following things:
- embeds an instance of rosserial_server (SerialSession)
- contains a RobotHW class (from the ros_control framework)
- publishes diagnostics

It is also where `base.launch` lives, which launches `jackal_node`, along with the description, controller, teleop stuff, and interpreters for IMU and Navsat data.

The remaining two executables produced (other than `jackal_node`):
- `simple_joy_teleop`, which may be used for direct joy -> cmd_drive translation, without a controller in the loop (though a parameter needs to be added to `jackal_node` to actually _disable_ the control loop in this scenario).
- `sixpair`, which is used for pairing the Playstation controller. Unfortunately, it presently requires root permissions to do this, which is a little awkward, but still doable.

@Shokoofeh to review

### CODE REVIEW

File: jackal_base/package.xml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  <run_depend>diff_drive_controller</run_depend>
+  <run_depend>jackal_msgs</run_depend>
+  <run_depend>nmea_msgs</run_depend>
+  <run_depend>nmea_navsat_driver</run_depend>
+  <run_depend>realtime_tools</run_depend>
+  <run_depend>rosserial_python</run_depend>
+  <run_depend>rosserial_server</run_depend>
+  <run_depend>sensor_msgs</run_depend>
+  <run_depend>teleop_twist_joy</run_depend>
+
```
Comment (shokoofeh) [inline jackal_base/package.xml:38]: i think we should add dependency to the imu_filter_madgwick" pkg.

File: jackal_base/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,39 @@
+cmake_minimum_required(VERSION 2.8.3)
+project(jackal_base)
+
+find_package(catkin REQUIRED COMPONENTS
+  controller_manager diagnostic_updater hardware_interface jackal_msgs roscpp roslaunch
+  roslint rosserial_server sensor_msgs teleop_twist_joy)
```
Comment (shokoofeh) [inline jackal_base/CMakeLists.txt]: "topic_tools" pks also needs to be found.

File: jackal_base/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+find_package(Boost)
+
+catkin_package()
+
+include_directories(include ${Boost_INCLUDE_DIR} ${catkin_INCLUDE_DIRS})
```
Comment (shokoofeh) [inline jackal_base/CMakeLists.txt]: include_directories(include ${Boost_INCLUDE_DIRS} ${catkin_INCLUDE_DIRS})

**Comment by mikepurvis** [comment]
Okay, issues addressed. Having looked a bit more, it's surprising to me that the manual find of topic_tools was necessary, given that `rosserial_serial` correctly specifies a CATKIN_DEPENDS on it. Anyhow, having the dependency here explicitly is okay for now.

**Pull request merged**